src/main/java/net/minecraftforge/items/itemhandlerobserver/IItemHandlerObserver.java

```diff
@@ -0,0 +1,17 @@
+package net.minecraftforge.items.itemhandlerobserver;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import javax.annotation.Nonnull;
+
+public interface IItemHandlerObserver
+{
+    void onStackInserted(IItemHandler handler, int slot, @Nonnull ItemStack newStack);
+
+    void onStackExtracted(IItemHandler handler, int slot, @Nonnull ItemStack newStack);
```

Comment:
Adding the oldstack or new/old amount might still have it's benefits. It avoids the need of using another local cache, should an observer need to forward the changed amounts as delta.